src/kernel/kernel_internal.h

```diff
@@ -41,20 +41,18 @@
  * to kthread_t) that can be used to externally manage the
  * thread like changing its priority or killing it.
  */
-typedef struct kthread_s
+typedef struct __attribute__((aligned (8))) kthread_s
 {
 	/** Pointer to the top of the stack, which contains the thread register state */
 	reg_block_t* stack_state;
-    struct {
-        /** Mirror of __interrupt_depth */
-        int interrupt_depth;
-        /** Mirror of __interrupt_sr */
-        int interrupt_sr;
-		#ifdef __NEWLIB__
-		/** Newlib reentrancy */
-		struct _reent *reent_ptr;
-		#endif
-    } tls;  ///< Thread-local storage
+	/** TLS Pointer */
+	void *tp_value;
+	/** Mirror of __interrupt_depth */
+	int interrupt_depth;
+	/** Mirror of __interrupt_sr  */
+	int interrupt_sr;
+	/** Newlib reentrancy */
+	struct _reent *reent_ptr;
```

Comment:
Any reason to remove the tis structure? I think It helped grouping them and providing also some implicit documentation.